### ISSUE TITLE
Which files should be used when smashing by species?

### ISSUE THREAD

**Issue by kurtwheeler** [AlexsLemonade/refinebio#963]
### Context

I'm currently working on #909, which is forcing me to review/rethink some of the ways we create, associate, and use files output by the salmon processor. One thing I've realized is that since a sample can belong to more than one experiment, it may have multiple smashable files which are each a slice of a different output from tximport. Currently our smash logic just uses the most recently created file. After discussing with @cansav09 we've identified that for experiment-level smash jobs we should be making sure to use files that came from the same tximport run. However we weren't able to figure out what to do when the smash job is at the species-level.

### Problem or idea

We need to figure out a good rule for picking a computed file to use for a given sample when smashing at the species level.

### Solution or next step

Unsure, but I imagine figuring out the answer to this question will involve @jaclyn-taroni 

### New Issue Checklist

- [x] The title is short and descriptive.
- [x] You have explained the context that led you to write this issue.
- [x] You have reported a problem or idea.
- [x] You have proposed a solution or next step.

**Comment by cansav09** [comment]
A potentially oversimplified answer for the species level question is to use the result file from the larger sample size. As we discussed, in the "ideal world" we'd probably want to run tximport with all the samples together, but that isn't practical.

**Comment by jaclyn-taroni** [comment]
If I’m understanding correctly: use the files from the same tximport run and the most recent set of files for species level smashing.

Do I have that right @cansav09? This is also what I was thinking.

**Comment by cansav09** [comment]
@jaclyn-taroni  Sort of. 

My thought is that for tximport experiments, you should use the tximport result from the same result. So if sample 1 is in both experiments A and B, (and thus has two tximport results) and the user is trying to get experiment A, definitely use the tximport result for sample 1 from Experiment A. Because we know from the tximport tests I did that the result will be quite drastically changed depending on the samples it's processed with. 

However, if the use is trying to get the samples from Experiment A and Experiment B, and thus you have to choose between two different results for sample 1, then my thought is that you should choose the tximport result from the bigger experiment. So say Experiment A is 500 sample and Experiment B is 100 samples, for sample 1, choose the Experiment A result. (my thought is that you would like all the samples to be comparable as possible. So pick the result that would help sample 1 relate to more samples?). 

In an ideal world, we'd want to re-run Experiment A and B's tximport to get the best results and make them the most comparable. But it sounds like this is not a practical thing. And I don't think the benefits of doing this outweigh the impracticality of it. 

In addition, for species level, if the same sample is in two experiments you are trying to smash together, my thought is you'd want to choose it's tximport file from the experiment with more samples.

I could be misunderstanding this, let me know what you guys think.

**Comment by jaclyn-taroni** [comment]
> My thought is that for tximport experiments, you should use the tximport result from the same result. So if sample 1 is in both experiments A and B, (and thus has two tximport results)

Does this happen in practice? My understanding is that it might not (see: https://github.com/AlexsLemonade/refinebio-docs/issues/19#issuecomment-423516346).

**Comment by kurtwheeler** [comment]
It does at least within the GEO side of things. [This sample](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM1692993) belongs to [GSE69108](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE69108) and [GSE69111](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE69111).

**Comment by jaclyn-taroni** [comment]
> It does at least within the GEO side of things.

Yes, but I don't think there is a "SuperSeries" of SRA. That's what I took away from our correspondence with EBI.  We should double check before going too deep down the rabbit hole here.

**Comment by cansav09** [comment]
> It does at least within the GEO side of things. [This sample](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM1692993) belongs to [GSE69108](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE69108) and [GSE69111](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE69111).

This is a microarray sample though.

**Comment by kurtwheeler** [comment]
Dammit, the database lied to me. We have that sample marked as  `technology='RNA-SEQ'` in the database. I guess that's another issue then... although our technology-detection code is really just best effort based on metadata.

**Comment by cansav09** [comment]
^ This isn't a thing you can determine definitively through an associated platform code?

**Comment by kurtwheeler** [comment]
Ok, so I found one that actually is RNA-Seq belonging to multiple experiments: https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM1693011

**Comment by jaclyn-taroni** [comment]
Yes, but at the moment, we should be ignoring RNA-seq samples on GEO as to avoid duplicates (via our microarray supported platforms list). So I'm still not sure that's relevant.

**Comment by kurtwheeler** [comment]
> Yes, but at the moment, we should be ignoring RNA-seq samples on GEO as to avoid duplicates (via our microarray supported platforms list). So I'm still not sure that's relevant.

Perhaps us not properly ignoring RNA-Seq samples on GEO is the bug then.

**Comment by jaclyn-taroni** [comment]
> Perhaps us not properly ignoring RNA-Seq samples on GEO is the bug then.

That would be a thing we want to know about / very good to track down.

**Comment by jaclyn-taroni** [comment]
> We have nothing that is even attempting to ignore RNA-seq samples on GEO.

Okay, thinking out loud a bit here: my understanding is that this most likely to be an issue where we would use our microarray supported platform (accession) list to generate a list of experiments and that list of experiments contains samples that are RNA-seq samples.

**Comment by kurtwheeler** [comment]
That hypothesis is consistent with what I've seen while looking into this. Haven't dug far enough to be able to say with 100% confidence though.

**Comment by kurtwheeler** [comment]
@cansav09 didn't mean to ignore:

> ^ This isn't a thing you can determine definitively through an associated platform code?

But the answer is yes and no. In an ideal world the answer would be yes, but sometime the samples are actually mislabeled. We actually open up CEL files later to make sure the headers match what platform we were told to expect. However you were right that in this case we should have detected it was microarray and when I tested locally we did properly detect the platform as long as we weren't surveying the super-series, which is broken I guess.

**Comment by cansav09** [comment]
> But the answer is yes and no. In an ideal world the answer would be yes, but sometime the samples are actually mislabeled.

Gotcha. Sounds like another case of things in GEO and other databases being disorganized and not streamlined.

**Comment by cgreene** [comment]
@kurtwheeler is this a thing we should close now? It seems like the recent fixes for not grabbing RNA-seq data from GEO will fix this, unless I'm misreading the discussion.
